src/protocol/assan_nrf24l01.c

```diff
@@ -0,0 +1,311 @@
+    {
+        temp=((s32)Channels[ch] * 0x1f1 / CHAN_MAX_VALUE + 0x5d9)<<3;
+        packet[2*ch]=temp>>8;
+        packet[2*ch+1]=temp;
+    }
+    NRF24L01_WriteReg(NRF24L01_07_STATUS, 0x70);        // Clear data ready, data sent, and retransmit
+    NRF24L01_FlushTx();
+    NRF24L01_WritePayload(packet, PACKET_SIZE);
+
+    // Check and adjust transmission power. We do this after
+    // transmission to not bother with timeout after power
+    // settings change -  we have plenty of time until next
+    // packet.
+    if (tx_power != Model.tx_power) {
+        //Keep transmit power updated
+        tx_power = Model.tx_power;
+        NRF24L01_SetPower(tx_power);
+    }
+}
+
+#define WAIT 0x80
+
+u16 ASSAN_callback()
+{
+    switch (state)
+    {
+    // Bind
+        case BIND0:
+            //Config RX @1M
+            NRF24L01_WriteReg(NRF24L01_05_RF_CH, RF_BIND_CHANNEL);
+            NRF24L01_SetBitrate(NRF24L01_BR_1M);                    // 1Mbps
+            NRF24L01_SetTxRxMode(RX_EN);
+            state++;
+        case BIND1:
+            //Wait for receiver to send the frames
+            if( NRF24L01_ReadReg(NRF24L01_07_STATUS) & BV(NRF24L01_07_RX_DR))
+            { //Something has been received
+                NRF24L01_ReadPayload(packet, PACKET_SIZE);
+                if(packet[19]==0x13)
+                { //Last frame received
+                    state++;
+                    state |= WAIT;
+                    //Switch to TX
+                    NRF24L01_SetTxRxMode(TXRX_OFF);
+                    NRF24L01_SetTxRxMode(TX_EN);
+                    //Prepare bind packet
+                    memset(packet,0x05,PACKET_SIZE-5);
+                    packet[15]=0x99;
+                    for(u8 i=0;i<4;i++)
+                        packet[16+i]=packet[23-i];
+                    packet_count=0;
+                    return 10000;
+                }
+            }
+            return 1000;
+        case BIND2|WAIT:
+            if(++packet_count == 27) // Wait 270ms in total...
+            {
+                packet_count = 0;
+                state &= ~WAIT;
+            }
+            return 10000;
+        case BIND2:
+            // Send 20 packets
+            packet_count++;
+            if(packet_count==20)
+                packet[15]=0x13;    // different value for last packet
+            NRF24L01_WritePayload(packet, PACKET_SIZE);
+            if(packet_count==20)
+            {
+                state++;
+                state |= WAIT;
+                packet_count = 0;
+            }
+            return 22520;
+        case DATA0|WAIT:
+            if(++packet_count == 217)
+                state &= ~WAIT;
+            return 10000;
+    // Normal operation
+        case DATA0:
+            // Bind Done
+            PROTOCOL_SetBindState(0);
+            NRF24L01_SetBitrate(NRF24L01_BR_250K);                  // 250Kbps
+            NRF24L01_SetTxRxMode(TXRX_OFF);
+            NRF24L01_SetTxRxMode(TX_EN);
```

Comment:
It is customary to add /* FALLTHROUGH */ if your case does not end with a break to notify others that you meant it.